pkg/agent/config.go

```diff
@@ -74,33 +86,44 @@ type AgentCmdFlags struct {
 	// YAML file.
 	ConfigFilePath string
 
-	// Period (--period, -p) is the time waited between scans.
+	// Period (--period, -p) is the time waited between scans. It takes
+	// precedence over the config field `period`.
 	Period time.Duration
 
-	// OneShot (--one-shot) flag causes agent to run once.
-	OneShot bool
-
-	// VenafiCloudMode (--venafi-cloud) determines which format to load for
-	// config and credential type.
+	// VenafiCloudMode (--venafi-cloud) turns on the Venafi Cloud Key Pair
+	// Service Account mode. Must be used in conjunction with
+	// --credentials-file.
 	VenafiCloudMode bool
 
-	// ClientID (--client-id) is the clientID in case of Venafi Cloud mode.
+	// ClientID (--client-id) is the clientID in case of Venafi Cloud Key Pair
+	// Service Account mode.
 	ClientID string
 
 	// PrivateKeyPath (--private-key-path) is the path for the service account
-	// private key in case of Venafi Cloud mode.
+	// private key in case of Venafi Cloud Key Pair Service Account mode.
 	PrivateKeyPath string
 
-	// CredentialsPath (--credentials-file, -k) is the path to the credentials )
-	// is where the agent will try to loads the credentials (Experimental).
+	// CredentialsPath (--credentials-file, -k) lets you specify the location of
+	// the credentials file. This is used for the Jetstack Secure OAuth and
+	// Venafi Cloud Key Pair Service Account modes. In Venafi Cloud Key Pair
+	// Service Account mode, you also need to pass --venafi-cloud.
 	CredentialsPath string
 
-	// OutputPath (--output-path) is where the agent will write data to instead
-	// of uploading to server.
+	// OneShot (--one-shot) is used for testing purposes. The agent will run
+	// once and exit. It is often used in conjunction with --output-path and/or
+	// --input-path.
+	OneShot bool
+
+	// OutputPath (--output-path) is used for testing purposes. In conjunction
+	// with --one-shot, it allows you to write the data readings to a file
+	// instead uploading them to the Venafi Cloud API.
 	OutputPath string
 
-	// InputPath (--input-path) is where the agent will read data from instead
-	// of gathering data from clusters.
+	// InputPath (--input-path) is used for testing purposes. In conjunction
+	// with --one-shot, it allows you to push manually crafted data readings (in
+	// JSON format) to the Venafi Cloud API without the need to connect to a
+	// Kubernetes cluster. See the jscp-testing-cli's README for more info:
+	// https://gitlab.com/venafi/vaas/applications/tls-protect-for-k8s/cloud-services/-/tree/master/jscp-testing-cli
```

Comment:
As indicated by Olu in https://github.com/jetstack/jetstack-secure/pull/575#issuecomment-2363335206, `--input-path` is being used internally at Venafi to test out the Venafi Cloud API. See the jscp-testing-cli's README for more info: https://gitlab.com/venafi/vaas/applications/tls-protect-for-k8s/cloud-services/-/tree/master/jscp-testing-cli

I've added "Used for testing purposes" to make it clearer that it isn't intended for general use.

I thought about hiding `--one-shot`, `--output-path`, and `--input-path` with

```go
c.PersistentFlags().MarkHidden("input-path")
c.PersistentFlags().MarkHidden("output-path")
c.PersistentFlags().MarkHidden("one-shot")
```

but I've decided to just mark them with "for testing purposes" since it achieves the same outcome: avoid confusion for end-users as to whether this flag would be useful to them or not.

@tfadeyi Does it make sense?